src/main/java/de/dennisguse/opentracks/io/file/exporter/InstantPostWorkoutExport.java

```diff
@@ -0,0 +1,40 @@
+package de.dennisguse.opentracks.io.file.exporter;
+
+import android.content.Context;
+import android.net.Uri;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.documentfile.provider.DocumentFile;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.OutputStream;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.io.file.ExportFile;
+import de.dennisguse.opentracks.io.file.TrackFileFormat;
+
+public class InstantPostWorkoutExport {
+
+    private static final String TAG = InstantPostWorkoutExport.class.getSimpleName();
+
+    public static void exportLastTrackToExternalStorage(Uri directoryUri, Context context, ContentProviderUtils contentProviderUtils, TrackFileFormat trackFileFormat, Track.Id trackId) {
+        DocumentFile directory = DocumentFile.fromTreeUri(context, directoryUri);
+
+        Track track = contentProviderUtils.getTrack(trackId);
+        DocumentFile file = ExportFile.getExportDocumentFile(trackId, trackFileFormat.getExtension(), directory, trackFileFormat.getExtension());
+
+        try (OutputStream outputStream = context.getContentResolver().openOutputStream(file.getUri())) {
+            TrackExportThread trackExportThread = new TrackExportThread(outputStream, context, trackFileFormat, track);
```

Comment:
TrackExportThread can be replaced by `new Thread(() -> trackFileFormat.newTrackExporter(context).writeTrack(track, outputStream)).start();`

Reply:
If i do it like you proposed it will only export 37bytes without any exception. The file does not contain all the data.